Update rubies (#1)

maxirmx opened the pull request:
* Resolve gcc 11 compatibility issue (https://github.com/masamitsu-murase/seven_zip_ruby/issues/36)
  This issue happens when newer versions of gcc are used where default C++ standard is 17.   There is an obvious mismatch in 
  7z code
```if (!mftRec.Parse(ByteBuf, Header.SectorSizeLog, numSectorsInRec, NULL, 0)) ```  instead of ```if (!mftRec.Parse(ByteBuf, Header.SectorSizeLog, numSectorsInRec, 0, NULL)) ``` 
  Newer versions of 7z have this fix. I have made a backport.
* Resolve Ruby 3.0.x/rubygems/bundler cache issue with 7z.so (p7zip converted to Ruby extension)
  The issue occurs with Ruby 3.0.x where 7z.so built outside of Ruby extension system was not considered for bundler cache
* Update GHA to run with new Rubies
* Use RSpec rake task

cc: @ronaldtse

This is a contribution from [Metanorma](https://www.metanorma.org/) ([GitHub](https://github.com/metanorma)), a product of [Ribose](https://www.ribose.com/) ([GitHub](https://github.com/riboseinc)).

ronaldtse committed:
Update rubies (#1)

* update GHA to run with multiple rubies
* feat: use RSpec rake task
* Resolve gcc 11 compatibility issue (https://github.com/masamitsu-murase/seven_zip_ruby/issues/36)
* Resolve Ruby 3.0.x/bundler cache issue with 7z.so (p7zip converted to Ruby extension)

Co-authored-by: Maxim [maxirmx] Samsonov <m.samsonov@computer.org>

ronaldtse commented:
(Thank you @maxirmx !)

@masamitsu-murase first of all, thank you for the fantastic gem! We (@riboseinc) have been using your seven_zip_ruby gem for our @metanorma and @fontist projects, and have a need to utilize the gem on the latest platforms and Ruby versions. I hope this PR of our contribution will be accepted. Thank you again!

sergiopantoja commented:
This is a wonderful contribution! I was having the `7z.so` issue as well and couldn't figure out the cause, but this branch solved it. Thank you.

maxirmx committed:
seven-zip gem created (#2)

maxirmx committed:
Added release workflow

maxirmx committed:
Fixed rubygems.org publishing issues

maxirmx committed:
Cleaned debug output and CI (#3)

maxirmx committed:
Niloufar fix dll path (#7)

Secure loading of libraries to prevent DLL preloading attacks
https://support.microsoft.com/en-us/help/2389418/secure-loading-of-libraries-to-prevent-dll-preloading-attacks

maxirmx committed:
Update p7zip sources to 16.02 and 7zip dlls to 19.00 (#8)

* Fix: Dll Loading

* Fix: Extract Path

* Update: p7zip 16.02

* Added a script to extract p7zip archive.

* Modified the program to support p7zip-16.02.

* Update: 7z1900 Dll's

* Patch: p7zip_16.02+dfsg-7.debian

* Modified p7zip/extract.sh to apply Debian Patch.

* Teeny version upgrade.

* Added Unit Test.

* Improved Unit Test.

* Added Unit Test.

* typo

* Added Unit Test.

* typo

* The code was improved.
Fixed an error message when decryption of wrong password failed.
Added taint marks to file paths in archives.

* Added the description of extrace.sh.

* Imported cut and pasted update of p7zip-16.02.
Besides, Added Tainted mark to generated Ruby String.

* The exception to be thrown was changed to an appropriate class.

* Merge completion

* Fixing issues

* Fixed invalid password tests

* rb_tainted_str_new (deprecated) --> rb_str_new

* Fixed merge issue

* Fixed compression level tests

* Consistent handling of invalid or missing passwords

Co-authored-by: niloufar <manew1n-jewr@yahoo.co.jp>

maxirmx committed:
Removed bin folder (#10) (#11)

maxirmx committed:
Fixed gcc 12 compatibility (#14)

* Fixed GCC 12 compatibility
* Added tests on Alpine
* Added tests on Ruby 3.2

maxirmx committed:
Updated README with 1.6.2 information

bkmgit commented:
@masamitsu-murase Any possibility of merging this or making the fork the official RubyGem version?

maxirmx commented:
> @masamitsu-murase Any possibility of merging this or making the fork the official RubyGem version?

@bkmgit 

We have to publish a fork
It is there:
https://github.com/fontist/seven_zip_ruby
https://rubygems.org/gems/seven-zip

Thank you

bkmgit committed:
Enable build on FreeBSD

CAMOBAP committed:
feat: migrate to shared workflow #18 #19

CAMOBAP committed:
Merge pull request #20 from fontist/chore/migrate-to-shared-workflow

feat: migrate to shared workflow #18 #19
